### PR TITLE
docs(aps): add firewall information

### PR DESCRIPTION
**Pull request by bene2k1**
### Description

Add information about firewall configuration to apple silicon docs

### CODE REVIEW

**Review by nerda-codes — APPROVED** [review]

File: bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx [hunk 1]
```diff
@@ -20,17 +20,60 @@ categories:
...
 
-However, some users that have recently upgraded their macOS are currently encountering a bug preventing them from effectively using the screen sharing feature. Error messages include `Connection failed to <M1s_address>` and `Make sure Screen Sharing or Remote Management (in the Sharing section of System Preferences) is enabled on the remote computer`.
+VNC (Virtual Network Computing) is a system that lets users remotely access and control another computer's desktop. For those using Scaleway's rented Mac mini M1, VNC can be activated by turning on either the "Screen Sharing" or "Remote Management" options in the system preferences. Once set up, you can connect to the Mac mini M1 from another device using its IP address or hostname through VNC clients like RealVNC, TightVNC, and others.
+
+However, some users who've updated their macOS have faced issues with this feature, receiving error messages like `Connection failed to <M1s_address>` and `Make sure Screen Sharing or Remote Management is enabled on the remote computer` under the Sharing section of System Preferences.
```
Comment (nerda-codes) [inline bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx:26]:
```suggestion
However, some users who have updated their macOS have faced issues with this feature, receiving error messages like `Connection failed to <M1s_address>` and `Make sure Screen Sharing or Remote Management is enabled on the remote computer` under the Sharing section of System Preferences. 
```

File: bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx [hunk 2]
```diff
@@ -20,17 +20,60 @@ categories:
...
 <Message type="note">
   This is a known issue that multiple Apple users have reported over time. Unfortunately, Apple's support team has not effectively addressed it yet.
 </Message>
 
-If you are experiencing this issue, you can try the following solutions:
+## Enable and Configure pf (Packet Filter)
+
+### Turn on the firewall
+
+macOS comes with a built-in firewall called `pf`. If you haven't already, enable the macOS firewall by going to **System Settings** -> **Network** -> **Firewall**. Toggle the switch to **On** to enable it.
```
Comment (nerda-codes) [inline bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx:40]:
```suggestion
macOS comes with a built-in firewall called `pf`. Enable the macOS firewall by going to **System Settings** -> **Network** -> **Firewall**. Toggle the switch to **On** to enable it.
```

File: bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx [hunk 3]
```diff
@@ -20,17 +20,60 @@ categories:
...
+    rdr-anchor "com.apple/*"
+    dummynet-anchor "com.apple/*"
+    anchor "com.apple/*"
+    load anchor "com.apple" from "/etc/pf.anchors/com.apple"
+    block drop in proto { tcp, udp } from <fail2ban> to any
+    ## Add these lines to block connections to port 5900 except from your static IP
+    pass in on en0 proto tcp from <static IP> to any port 5900
+    block in on en0 proto tcp from any to any port 5900
+    ```
+    In the example above, `en0` should be replaced with the network interface you are using (you may discover the name of your network interface name using the `ifconfig` command), and `<static IP>` should be the static IP address you are allowing. Port 5900 is typically used for VNC/screen sharing services.
```
Comment (nerda-codes) [inline bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx:65]:
```suggestion
    In the example above, `en0` should be replaced with the network interface you are using (you may discover the name of your network interface using the `ifconfig` command), and `<static IP>` should be the static IP address you are allowing. Port 5900 is typically used for VNC/screen sharing services.
```

File: bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx [hunk 4]
```diff
@@ -20,17 +20,60 @@ categories:
...
+5. Save and close the file (in `nano`, it's **Control-X** to close, **Y** to confirm changes, and **Enter** to save).
```
Comment (nerda-codes) [inline bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx:66]:
```suggestion
5. Save and close the file (if you are using `nano`: type **Y** to confirm your changes, **Enter** to save, and **Control-X** to close).
```

**Review by nerda-codes — COMMENTED** [review]
The spelling for "Mac Mini M1" is not always consistent :)

File: bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx [hunk 5]
```diff
@@ -20,17 +20,60 @@ categories:
...
+6. Run `sudo pfctl -e -f /etc/pf.conf` to load the new `pf` configuration. You might need to restart your computer for changes to take effect.
+7. Initiate a Screen Sharing session from the approved IP address and ensure it connects. Also, test from a different IP address to confirm it does not allow the connection.
+  <Message type="note">
+    - Any changes in your network configuration or the external static IP might require updates to this configuration. Keep this in mind, and adjust as necessary.
+    - Remember that this setup can cause issues if the static IP changes for some reason. You can access your Mac mini using SSH to update the configuration.
+  </Message>
+
+## Restaring screen sharing via SSH
```
Comment (jcirinosclwy) [inline bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx:74]:
```suggestion
## Restarting screen sharing via SSH
```

File: bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx (hunk 1 above)
Comment (jcirinosclwy) [inline bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx:26]:
```suggestion
However, some users who have updated their macOS have faced issues with this feature, receiving error messages like `Connection failed to <M1s_address>` and `Make sure Screen Sharing or Remote Management is enabled on the remote computer` under the Sharing section of System Preferences. 
```

File: bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx [hunk 6]
```diff
@@ -20,17 +20,60 @@ categories:
...
+Upon further analysis, we found a design flaw in macOS 13 and 14, where brute-force attacks performed by malicious actors could result in the service halting. [Find out more about this issue on our blog](https://www.scaleway.com/en/blog/m1-how-a-design-flaw-slowed-down-apples-vnc/).
+
+As a precaution, every Mac mini comes with a preinstalled `fail2ban` service. For added security and to ensure VNC service reliability, you can restrict access to macOS Screen Sharing to certain static IPs. This can be done by tweaking your network settings and using macOS's inbuilt firewall, `pf` (Packet Filter), to manage incoming connections.
 
 <Message type="note">
   This is a known issue that multiple Apple users have reported over time. Unfortunately, Apple's support team has not effectively addressed it yet.
 </Message>
 
-If you are experiencing this issue, you can try the following solutions:
+## Enable and Configure pf (Packet Filter)
```
Comment (jcirinosclwy) [inline bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx:36]:
```suggestion
## Enable and configure pf (Packet Filter)
```

File: bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx [hunk 7]
```diff
@@ -20,17 +20,60 @@ categories:
...
+VNC (Virtual Network Computing) is a system that lets users remotely access and control another computer's desktop. For those using Scaleway's rented Mac mini M1, VNC can be activated by turning on either the "Screen Sharing" or "Remote Management" options in the system preferences. Once set up, you can connect to the Mac mini M1 from another device using its IP address or hostname through VNC clients like RealVNC, TightVNC, and others.
+
+However, some users who've updated their macOS have faced issues with this feature, receiving error messages like `Connection failed to <M1s_address>` and `Make sure Screen Sharing or Remote Management is enabled on the remote computer` under the Sharing section of System Preferences. 
+
+Upon further analysis, we found a design flaw in macOS 13 and 14, where brute-force attacks performed by malicious actors could result in the service halting. [Find out more about this issue on our blog](https://www.scaleway.com/en/blog/m1-how-a-design-flaw-slowed-down-apples-vnc/).
```
Comment (scwgoire) [inline bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx:28]: let's talks about macOS in general, no need to tell about the version, we don't support macOS 14 yet btw

File: bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx [hunk 8]
```diff
@@ -20,17 +20,60 @@ categories:
...
+
+As a precaution, every Mac mini comes with a preinstalled `fail2ban` service. For added security and to ensure VNC service reliability, you can restrict access to macOS Screen Sharing to certain static IPs. This can be done by tweaking your network settings and using macOS's inbuilt firewall, `pf` (Packet Filter), to manage incoming connections.
```
Comment (scwgoire) [inline bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx:30]: Maybe we could explain what fail2ban is here (parses logs, notices auth failures, and configures firewall)

**Review by jcirinosclwy — APPROVED** [review]

File: bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx (hunk 2 above)
Comment (scwgoire) [inline bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx]: Should be already enabled by macOS "customization" (fail2ban)

File: bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx [hunk 9]
```diff
@@ -20,17 +20,60 @@ categories:
...
+   ```
+3. Edit the `pf.conf` file with a command like `sudo nano /etc/pf.conf`.
+4. Add the following lines to write rules that block all incoming connections on the Screen Sharing port, except for the desired IP.
+    ```
+    scrub-anchor "com.apple/*"
+    nat-anchor "com.apple/*"
+    rdr-anchor "com.apple/*"
+    dummynet-anchor "com.apple/*"
+    anchor "com.apple/*"
+    load anchor "com.apple" from "/etc/pf.anchors/com.apple"
```
Comment (scwgoire) [inline bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx:59]: not sure what these anchor do, maybe worth explaining?

File: bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx [hunk 10]
```diff
@@ -20,17 +20,60 @@ categories:
...
+    block drop in proto { tcp, udp } from <fail2ban> to any
```
Comment (scwgoire) [inline bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx:60]: Any fail2ban-related rule may be dynamically loaded at machine startup, did you test this including a machine reboot?

File: bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx [hunk 11]
```diff
@@ -20,17 +20,60 @@ categories:
...
+    ## Add these lines to block connections to port 5900 except from your static IP
+    pass in on en0 proto tcp from <static IP> to any port 5900
+    block in on en0 proto tcp from any to any port 5900
+    ```
+    In the example above, `en0` should be replaced with the network interface you are using (you may discover the name of your network interface name using the `ifconfig` command), and `<static IP>` should be the static IP address you are allowing. Port 5900 is typically used for VNC/screen sharing services.
+5. Save and close the file (in `nano`, it's **Control-X** to close, **Y** to confirm changes, and **Enter** to save).
+6. Run `sudo pfctl -e -f /etc/pf.conf` to load the new `pf` configuration. You might need to restart your computer for changes to take effect.
```
Comment (scwgoire) [inline bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx:67]: pf should be already enabled so the `-e` shouldn't be needed I guess?

also, if the a restart is needed then it is not the right command line.

File: bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx [hunk 12]
```diff
@@ -20,17 +20,60 @@ categories:
...
-If you are experiencing this issue, you can try the following solutions:
+## Enable and Configure pf (Packet Filter)
+
+### Turn on the firewall
+
+macOS comes with a built-in firewall called `pf`. If you haven't already, enable the macOS firewall by going to **System Settings** -> **Network** -> **Firewall**. Toggle the switch to **On** to enable it.
+
+### Configure the firewall
+
+You have to create a custom rule set for `pf` to only allow incoming Screen Sharing connections from your static IP.
```
Comment (scwgoire) [inline bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx:44]: Maybe add a note "we recommend you do this through SSH to avoid loosing connection in case ... blah blah"

File: bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx [hunk 13]
```diff
@@ -39,19 +82,21 @@ If you are experiencing this issue, you can try the following solutions:
 
 2. Run the following command to forcefully terminate Screen Sharing:
     ```
-    sudo killall screensharing
+    sudo killall screensharingd
```
Comment (scwgoire) [inline bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx:85]: screensharingd will be automagically restarted at next incoming connection.

File: bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx [hunk 14]
```diff
@@ -39,19 +82,21 @@ If you are experiencing this issue, you can try the following solutions:
...
     ```
 
 3. Run the following command to load the `screensharing.plist` file responsible for Screen Sharing:
```
Comment (scwgoire) [inline bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx:88]: I'm not sure 3 and 4 are necessary to restart screensharingd. It most probably is required only to start or stop it, but if it has been started (as it should be by our macOS installation) then there should be no need for it.

Comment (scwgoire) [reply bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx:88]: requires testing I guess

File: bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx [hunk 15]
```diff
@@ -63,3 +108,7 @@ Make sure that either "Screen sharing" or "Remote management" are enabled on you
 ### Verify that incoming VNC connections are not blocked by the firewall
 
 Make sure that your Mac Mini M1 firewall allows incoming connections for your VNC client. Go to **System Settings** and then, depending on the OS version, **Firewall** will be located in either **Network** or **Security & Privacy**.
+
+### Further information
+
+* Official Apple documentation: [Block connections to your Mac with a firewall](https://support.apple.com/en-gb/guide/mac-help/mh34041/14.0/mac/14.0)
```
Comment (scwgoire) [inline bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx:114]: I think this is unrelated to this doc because it explains how to block all traffic to a service, with no possibility to filter good/bad sources.

File: bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx (hunk 14 above)
Comment (bene2k1) [reply bare-metal/apple-silicon/troubleshooting/cant-connect-using-vnc.mdx]: Indeed, it restarts automatically
